servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -441,54 +462,125 @@ public Completable closeAsyncGracefully() {
         return usedHosts.stream().map(Host::asEntry).collect(toList());
     }
 
-    private static final class Host<Addr, C extends ListenableAsyncCloseable> implements ListenableAsyncCloseable {
-        private static final ConnState EMPTY_CONN_STATE = new ConnState(EMPTY_ARRAY, State.ACTIVE);
-        private static final ConnState CLOSED_CONN_STATE = new ConnState(EMPTY_ARRAY, State.CLOSED);
+    static final class HealthCheckConfig {
+        private final Executor executor;
+        private final Duration healthCheckInterval;
+        private final int failedThreshold;
+
+        HealthCheckConfig(final Executor executor, final Duration healthCheckInterval, final int failedThreshold) {
+            this.executor = executor;
+            this.healthCheckInterval = healthCheckInterval;
+            this.failedThreshold = failedThreshold;
+        }
+    }
+
+    private static final class Host<Addr, C extends LoadBalancedConnection> implements ListenableAsyncCloseable {
+        private static final Object STATE_ACTIVE = new Object();
+        private static final Object STATE_EXPIRED = new Object();
+        private static final Object STATE_CLOSED = new Object();
```

Comment:
As a matter of fact, with the latest suggestion to keep a counter in the active state object, this will not be a feasible option. `STATE_EXPIRED`, and `STATE_CLOSED` would be the only constants and e.g. `ActiveState` and `HealthCheck` will be new instances upon each update to the state.

Reply:
heap dump shows class names, so `ActiveState` and `HealthCheck` will be readable, but `Object` won't. Can we have a enum for `expired` and `closed`?